ablateLibrary/domain/modifiers/distributeWithGhostCells.cpp

```diff
@@ -0,0 +1,19 @@
+#include "distributeWithGhostCells.hpp"
+#include <utilities/petscError.hpp>
+
+ablate::domain::modifier::DistributeWithGhostCells::DistributeWithGhostCells(int ghostCellDepthIn) : ghostCellDepth(ghostCellDepthIn < 1 ? 1 : ghostCellDepthIn) {}
+void ablate::domain::modifier::DistributeWithGhostCells::Modify(DM &dm) {
+    // Make sure that the flow is setup distributed
+    DM dmDist;
+    // create any ghost cells that are needed
+    DMSetBasicAdjacency(dm, PETSC_TRUE, PETSC_FALSE) >> checkError;
+    DMPlexDistribute(dm, ghostCellDepth, NULL, &dmDist) >> checkError;
```

Comment:
Note that this is one option for having overlap, but another is to just add the call to `DMPlexDistributeOverlap()` after a regular distribution.